pulsar-metadata/src/main/java/org/apache/pulsar/metadata/bookkeeper/PulsarLedgerIdGenerator.java

```diff
@@ -287,4 +288,16 @@ private static String createLedgerPrefix(String ledgersPath, String idGenZnodeNa
         return ledgerIdGenPath + "/" + "ID-";
     }
 
+    //If the config rootPath when use zk metadata store, it will append rootPath as the prefix of the path.
+    //So when we get the path from the stat, we should truncate the rootPath.
+    private String handleTheDeletePath(String path) {
+        if (store instanceof ZKMetadataStore) {
+            String rootPath = ((ZKMetadataStore) store).getRootPath();
+            if (rootPath == null) {
+                return path;
+            }
+            return path.replaceFirst(rootPath, "");
+        }
+        return path;
+    }
```

Comment:
Could we make this in the constructor to avoid calling `replaceFirst` every time?

Reply:
It's not a fixed path, it looks like we can't avoid replacing operation.

Reply:
> It's not a fixed path, it looks like we can't avoid replacing operation.

+1